libsycl/include/sycl/__impl/platform.hpp

```diff
@@ -15,17 +15,65 @@
 #ifndef _LIBSYCL___IMPL_PLATFORM_HPP
 #define _LIBSYCL___IMPL_PLATFORM_HPP
 
+#include <sycl/__impl/backend.hpp>
 #include <sycl/__impl/detail/config.hpp>
+#include <sycl/__impl/detail/obj_utils.hpp>
+#include <sycl/__impl/info/platform.hpp>
+
+#include <memory>
+#include <vector>
 
 _LIBSYCL_BEGIN_NAMESPACE_SYCL
 
+namespace detail {
+class platform_impl;
+} // namespace detail
+
+/// \brief SYCL 2020 platform class (4.6.2.) encapsulating a single SYCL
+/// platform on which kernel functions may be executed.
 class _LIBSYCL_EXPORT platform {
 public:
-  /// Constructs a SYCL platform which contains the default device.
-  platform();
+  platform(const platform &rhs) = default;
```

Comment:
Since the SYCL synopsis for `platform` doesn't explicitly list the copy/move related functions, I think it would be useful to include a comment referencing the common reference semantic requirements. This comment might be better placed elsewhere, I suggested this location because the relevant functions appear to be (intentionally or unintentionally) grouped together here.
```suggestion
  // The platform class provides the common reference semantics (SYCL 2020 4.5.2).
  platform(const platform &rhs) = default;
```